doc/source/whatsnew/v0.24.0.txt

```diff
@@ -208,6 +208,8 @@ Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 - A newly constructed empty :class:`DataFrame` with integer as the ``dtype`` will now only be cast to ``float64`` if ``index`` is specified (:issue:`22858`)
+- The `.str`-accessor does not work for anything other than string data anymore. Any other uses (like ``bytes``, for example)
```

Comment:
can you reword a bit. something like

```
The `.str`-accessor will raise on any non-(string or unicode) ``object`` dtype data. Previously this would accept ``bytes`` for example.
```

Reply:
ok, then i think we maybe need to just show a warning for non-string/unicode uses. I guess this would generalize to calling anything-other-than-decode on bytes as well (though should raise on that).

So what i'll suggest is that we do an infer pass at the very beginning, rejecting the obvious things, then pass that inferred string around so that the methods can decide what to do, and have several canned responsed, eg.. a standard warning, or an error (for the bytes things).

Reply:
you *may* actually want to create a class object around which has the pointers to the data and the inferred state. this is similar to how we handled infer_dtype dispatching.